### PR TITLE
Add Elasticsearch service support and add getting and filtering produ…

### PR DESCRIPTION
**Pull request by nurikkk99**

### CODE REVIEW

File: customer-service/src/main/java/com/epam/customerservice/dto/GoodsType.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.epam.customerservice.dto;
+
+public enum GoodsType {
```
Comment (yelis913) [inline customer-service/src/main/java/com/epam/customerservice/dto/GoodsType.java:3]: Not an issue for an educational project, but in production an enum would be a constraint. What if you need to add a type later? I recommend just a String.

File: customer-service/src/main/java/com/epam/customerservice/service/ProductService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            response = client.search(request, RequestOptions.DEFAULT);
+            SearchHit[] searchHits = response.getHits().getHits();
+            List<ProductDto> products = new ArrayList<>(searchHits.length);
+
+            for(SearchHit hit : searchHits) {
+                products.add(objectMapper.readValue(hit.getSourceAsString(), ProductDto.class));
+            }
+            return products;
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (yelis913) [inline customer-service/src/main/java/com/epam/customerservice/service/ProductService.java]: I'd throw an exception here without logging. Let the REST controller or the ExceptionHandler decide what to do with it.

File: customer-service/src/main/java/com/epam/customerservice/dto/GoodsType.java (hunk 1 above)
Comment (nurikkk99) [reply customer-service/src/main/java/com/epam/customerservice/dto/GoodsType.java:3]: Got it!

File: customer-service/src/main/java/com/epam/customerservice/service/ProductService.java (hunk 2 above)
Comment (nurikkk99) [reply customer-service/src/main/java/com/epam/customerservice/service/ProductService.java]: Fixed it

**Pull request merged**